From d4862c214eaa190a677ff5602d3b5911977c1d21 Mon Sep 17 00:00:00 2001
From: Caleb Ethridge <caleb.ethridge@analog.com>
Date: Mon, 17 Aug 2026 09:16:15 -0400
Subject: [PATCH] sc598: Enlarge u-boot flash partition for secure boot

The secure boot stage-1 LDR carries OP-TEE and TF-A along with the SPL so
the boot ROM authenticates them together, which no longer fits the 256 KiB
u-boot-spl partition. Give the u-boot-spl and u-boot partitions 1 MiB each
and move the kernel and rootfs to make room.

Signed-off-by: Caleb Ethridge <caleb.ethridge@analog.com>
---
 arch/arm64/boot/dts/adi/sc598-som-revD.dtsi | 14 +++++++-------
 arch/arm64/boot/dts/adi/sc598-som-revE.dtsi | 14 +++++++-------
 2 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/arch/arm64/boot/dts/adi/sc598-som-revD.dtsi b/arch/arm64/boot/dts/adi/sc598-som-revD.dtsi
index 8189095325ff..7dd104da41ec 100644
--- a/arch/arm64/boot/dts/adi/sc598-som-revD.dtsi
+++ b/arch/arm64/boot/dts/adi/sc598-som-revD.dtsi
@@ -81,22 +81,22 @@ partitions {
 
 			partition@0 {
 				label = "u-boot-spl";
-				reg = <0x00000000 0x40000>;
+				reg = <0x00000000 0x100000>;
 			};
 
-			partition@40000 {
+			partition@100000 {
 				label = "u-boot";
-				reg = <0x00040000 0xc0000>;
+				reg = <0x00100000 0x100000>;
 			};
 
-			partition@100000 {
+			partition@200000 {
 				label = "kernel";
-				reg = <0x00100000 0xf00000>;
+				reg = <0x00200000 0xf00000>;
 			};
 
-			partition@1000000 {
+			partition@1100000 {
 				label = "rootfs";
-				reg = <0x01000000 0x3000000>;
+				reg = <0x01100000 0x2f00000>;
 			};
 		};
 	};
diff --git a/arch/arm64/boot/dts/adi/sc598-som-revE.dtsi b/arch/arm64/boot/dts/adi/sc598-som-revE.dtsi
index fd87d5cc3c66..c7f938f6cce1 100644
--- a/arch/arm64/boot/dts/adi/sc598-som-revE.dtsi
+++ b/arch/arm64/boot/dts/adi/sc598-som-revE.dtsi
@@ -79,22 +79,22 @@ partitions {
 
 			partition@0 {
 				label = "u-boot-spl";
-				reg = <0x00000000 0x40000>;
+				reg = <0x00000000 0x100000>;
 			};
 
-			partition@40000 {
+			partition@100000 {
 				label = "u-boot";
-				reg = <0x00040000 0xc0000>;
+				reg = <0x00100000 0x100000>;
 			};
 
-			partition@100000 {
+			partition@200000 {
 				label = "kernel";
-				reg = <0x00100000 0x2000000>;
+				reg = <0x00200000 0x2000000>;
 			};
 
-			partition@2100000 {
+			partition@2200000 {
 				label = "rootfs";
-				reg = <0x02100000 0x5f00000>;
+				reg = <0x02200000 0x5e00000>;
 			};
 		};
 	};
-- 
2.34.1

